### PR TITLE
Set permissions for GitHub actions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -13,6 +13,9 @@ env:
 jobs:
 
   lint:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
+      pull-requests: read  # for golangci/golangci-lint-action to fetch pull requests
```
Comment (AkihiroSuda) [inline .github/workflows/validate.yml:18]: Can be another PR, but eventually we should set permissions explicitly for all the jobs, not just `lint`?

Comment (naveensrinivasan) [reply .github/workflows/validate.yml:18]: Yes, that can be done.

**Comment by kolyshkin** [comment]
@naveensrinivasan so why are we specifically targeting lint job?

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan so why are we specifically targeting lint job?

That was the easiest to address.

**Comment by kolyshkin** [comment]
So, I think that something like this should be done instead: https://github.com/opencontainers/runc/pull/3445

**Comment by kolyshkin** [comment]
Superceded by #3445
